### PR TITLE
Dev/product form

### PR DESCRIPTION
**Pull request by BaidaaAlbzoor**

### CODE REVIEW

File: src/app/login/login.component.html [hunk 1]
```diff
@@ -24,18 +24,17 @@ <h4 class="version">v{{version}}</h4>
         <br/>
```
Comment (Moathomar-cp) [inline src/app/login/login.component.html]: why did you changed to login file ? 
its ok and NOT MERGE BLOCKING !  but i want to know why :D

File: src/app/product-dialog/product-dialog.component.html [hunk 2]
```diff
@@ -0,0 +1,130 @@
+<div class="mat-container">
```
Comment (Moathomar-cp) [inline src/app/product-dialog/product-dialog.component.html]: Nothing Specific. but as a reminder we still need another iteration here to add [(ngModel)]
NOT MERGE BLOCKING

File: src/app/products/products.component.ts [hunk 3]
```diff
@@ -2,17 +2,25 @@ import { Component, OnInit } from '@angular/core';
...
   templateUrl: './products.component.html',
   styleUrls: ['./products.component.scss']
 })
 export class ProductsComponent implements OnInit {
-
+  searchText: string;
   products: Product[];
-  constructor(private spinnerService: Ng4LoadingSpinnerService) { }
+  dialogRef: MatDialogRef<ProductDialogComponent>;
+  dialogRef2: MatDialogRef<CategoryFormComponent>;
```
Comment (Moathomar-cp) [inline src/app/products/products.component.ts]: its better to name them with more descriptive names like categoryDialogRef

File: src/app/category-form/category-form.component.html [hunk 4]
```diff
@@ -0,0 +1,23 @@
+<div class="mat-container">
+  <h1 mat-dialog-title>Add Category
+    <button class="close">X</button>
```
[the remark this answers is not in the document]
Comment (BaidaaAlbzoor) [reply src/app/category-form/category-form.component.html]: fixed

File: src/app/products/products.component.ts [hunk 5]
```diff
@@ -27,4 +35,30 @@ export class ProductsComponent implements OnInit {
...
+    this.dialogRef = this.dialog.open(ProductDialogComponent, {
+      width: '600px',
+      height: '700px',
+
+    });
+
+  }
+
+
+  openDialog2() {
```
[the remark this answers is not in the document]
Comment (BaidaaAlbzoor) [reply src/app/products/products.component.ts]: Done

File: src/app/products/products.component.ts [hunk 6]
```diff
@@ -27,4 +35,30 @@ export class ProductsComponent implements OnInit {
     this.products = PRODUCTS;
   }
 
+  openDialog() {
```
[the remark this answers is not in the document]
Comment (BaidaaAlbzoor) [reply src/app/products/products.component.ts]: Done

File: src/app/login/login.component.html (hunk 1 above)
Comment (BaidaaAlbzoor) [reply src/app/login/login.component.html]: It gave me errors when i pulled master br, so I commented this part of code just to continue my work in the form and refer to this error after I finish to fix it. (Stupid solution  :D ).

File: src/app/product-dialog/product-dialog.component.html (hunk 2 above)
Comment (BaidaaAlbzoor) [reply src/app/product-dialog/product-dialog.component.html]: I had modified it and did not find the code after I did the pull request! so I modified it again

File: src/app/app.module.ts [hunk 7]
```diff
@@ -52,7 +54,9 @@ import { CategoriesModule } from './categories/categories.module';
     ExpensesListModule,
     Ng4LoadingSpinnerModule.forRoot(),
     ProductsModule,
+    CategoryFormModule,
     ProvidersModule,
+    AppRoutingModule,
```
Comment (Moathomar-cp) [inline src/app/app.module.ts]: @BaidaaAlbzoor  AppRoutingModule, was already imported , and it must kept last one as we learned from routing issue

**Pull request merged**